miner/bid_simulator.go

```diff
@@ -617,15 +621,35 @@ func (b *bidSimulator) simBid(interruptCh chan int32, bidRuntime *BidRuntime) {
 		}
 	}
 
-	bidRuntime.packReward(b.config.ValidatorCommission)
+	// check if bid reward is valid
+	{
+		bidRuntime.packReward(b.config.ValidatorCommission)
+		if !bidRuntime.validReward() {
+			err = errors.New("reward does not achieve the expectation")
+			return
+		}
+	}
 
-	// return if bid is invalid, reportIssue issue to mev-sentry/builder if simulation is fully done
-	if !bidRuntime.validReward() {
-		err = errors.New("reward does not achieve the expectation")
-		return
+	// check if bid gas price is lower than min gas price
+	{
+		baseFee := bidRuntime.env.header.BaseFee
+		bidGasUsed := uint64(0)
+		bidGasFee := new(big.Int)
+
+		for i, receipt := range bidRuntime.env.receipts {
+			bidGasUsed += receipt.GasUsed
+			effectiveGasPrice := new(big.Int).Add(baseFee, bidRuntime.env.txs[i].EffectiveGasTipValue(baseFee))
```

Comment:
what is the usage of `effectiveGasPrice`?

Reply:
for calculating txFee. If using the balance of system address, there maybe some transfer tx as tips to system.

Reply:
yeah, that is what we want I think, sending BNB to system address is also valid.

Reply:
fixed